.github/workflows/ci.yml

```diff
@@ -325,6 +325,46 @@ jobs:
           flag-name: backend-integration
           parallel: true
 
+  backend-tests-functional:
+    if: |
+      always() && !cancelled() &&
+      !contains(needs.*.result, 'failure') &&
+      !contains(needs.*.result, 'cancelled') &&
+      needs.files-changed.outputs.backend == 'true'
+    needs: ["files-changed", "yaml-lint", "python-lint"]
+    runs-on:
+      group: "huge-runners"
+    timeout-minutes: 30
+    env:
+      INFRAHUB_DB_TYPE: neo4j
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v4"
+        with:
+          submodules: true
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - name: "Setup git credentials"
+        run: "git config --global user.name 'Infrahub' && \
+              git config --global user.email 'infrahub@opsmill.com' && \
+              git config --global --add safe.directory '*' && \
+              git config --global credential.usehttppath true && \
+              git config --global credential.helper /usr/local/bin/infrahub-git-credential"
+      - name: "Setup Python environment"
+        run: |
+          poetry config virtualenvs.create true --local
+          poetry env use 3.12
+      - name: "Install dependencies"
+        run: "poetry install --no-interaction --no-ansi"
+      - name: "Mypy Tests"
+        run: "poetry run invoke backend.mypy"
+      - name: "Pylint Tests"
+        run: "poetry run invoke backend.pylint"
```

Comment:
Given how slow pylint is I think if we add it here we can remove it from the integration tests.